docs-ref-services/key-vault.md

```diff
@@ -142,18 +111,140 @@ operation = client.vaults.create_or_update(
     }
 )
 
+# Run the operation
 vault = operation.result()
+
+# Show the URL of the created vault
 print(f'New vault URI: {vault.properties.vault_uri}')
 ```
 
 > [!div class="nextstepaction"]
 > [Explore the Management APIs](/python/api/overview/azure/keyvault/management)
 
-## Samples
-* [Manage Azure Key Vaults][1] 
-* [Azure Key Vault recovery][2]
+## Client library
+
+Install the library with `pip install azure-keyvault`.
+
+To access a key vault, an application must first authenticate with the Key Vault service. Authentication is best done using [Azure Managed Identities](/azure/active-directory/managed-identities-azure-resources/overview) (part of Azure Active Directory), and can also also be done with a service principal.
+
+To use Managed Identities, acquire a `DefaultAzureCredential` object. (To run this code on a developer workstation, see [Using Managed Identities locally](#using-managed-identities-locally)).
+
+```python
+# Authentication with Managed Identities
+from azure.identity import DefaultAzureCredential
+
+# Acquire default Azure credentials, which are useful in most circumstances.
+credentials = DefaultAzureCredential()
+```
+
+To use a service principal, acquire a `ServicePrincipalCredential` object. Review [Authenticate with the Azure SDK for Python](/azure/python/python-sdk-azure-authenticate) for information on obtaining a tenant ID, a client ID, and application secret.
+
+```python
+# Authentication with a service principal
+from azure.common.credentials import ServicePrincipalCredentials
+
+# Replace the values with your specific IDs
+client_id = "your_client_id"
+tenant_id = "your_tenant_id"
+app_secret = "your_app_secret"
+
+credentials = ServicePrincipalCredentials(client_id, app_secret, tenant_id)
+```
+
+With the `credentials` variable initialized, you can then use the following code to create a secret named `ExampleSecret` in an Azure Key Vault named `msdocs-vault01` (as created using the code in the [management library](#management-library) section.
+
+```python
+# You can place this import statement earlier in your code.
+from azure.keyvault.secrets import SecretClient
+
+# The URL of your Key Vault obtained from the Azure portal, which ends in vault.azure.net.
+# This example uses https://msdocs-vault01.vault.azure.net/
+key_vault_uri = "https://msdocs-vault01.vault.azure.net/"
+
+# Obtain the client object
+secret_client = SecretClient(key_vault_uri, credentials)
+
+# Create a secret. In production code deployed to a public server, such values
+# would not appear in source code.
+secret_name = "ExampleSecret"
+secret_client.set_secret(secret_name, "value5502")
```

Comment:
... which is to say, this will raise an error if `credentials` is `ServicePrincipalCredentials`.